// @angular/core/core.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
export function ApplicationModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.ɵe,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵConsole}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.Injector}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ErrorHandler}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ComponentFactoryResolver}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationInitStatus}}}],
      lifecycleHooks:[1]}},useClass:{reference:i0.ɵe,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵConsole}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Injector}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ErrorHandler}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ComponentFactoryResolver}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ApplicationInitStatus}}}],lifecycleHooks:[1]},
      useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
      deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵConsole}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.Injector}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ErrorHandler}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ComponentFactoryResolver}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationInitStatus}}}],
      multi:false},module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ApplicationRef}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i0.ɵe}},deps:(undefined as any),multi:false},
      module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ApplicationInitStatus,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.APP_INITIALIZER}}}],lifecycleHooks:([] as any[])}},
      useClass:{reference:i0.ApplicationInitStatus,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.APP_INITIALIZER}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.APP_INITIALIZER}}}],
      multi:false},module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.Compiler,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i0.Compiler,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
      useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),multi:false},
      module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.APP_ID}},
      useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵf,
          diDeps:([] as any[])},useExisting:(undefined as any),deps:([] as any[]),
      multi:false},module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.IterableDiffers}},
      useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵk,
          diDeps:([] as any[])},useExisting:(undefined as any),deps:([] as any[]),
      multi:false},module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.KeyValueDiffers}},
      useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵl,
          diDeps:([] as any[])},useExisting:(undefined as any),deps:([] as any[]),
      multi:false},module:{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.LOCALE_ID}},
      useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵm,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i0.LOCALE_ID}}}]},useExisting:(undefined as any),
      deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
          token:{identifier:{reference:i0.LOCALE_ID}}}],multi:false},module:{reference:i0.ApplicationModule,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ApplicationRef}}}],lifecycleHooks:([] as any[])}}],
      modules:[{reference:i0.ApplicationModule,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ApplicationRef}}}],
          lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),exportedPipes:([] as any[])},
      {summaryKind:3,type:{reference:i0.ɵe,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgZone}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ɵConsole}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Injector}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ErrorHandler}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ComponentFactoryResolver}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ApplicationInitStatus}}}],lifecycleHooks:[1]}},
      {summaryKind:3,type:{reference:i0.ApplicationInitStatus,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.APP_INITIALIZER}}}],
          lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.Compiler,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function ɵeNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵe,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵConsole}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Injector}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ErrorHandler}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ComponentFactoryResolver}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ApplicationInitStatus}}}],lifecycleHooks:[1]}}];
}
export function ApplicationInitStatusNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ApplicationInitStatus,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.APP_INITIALIZER}}}],
      lifecycleHooks:([] as any[])}}];
}
export function CompilerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.Compiler,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function SystemJsNgModuleLoaderNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.SystemJsNgModuleLoader,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Compiler}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.SystemJsNgModuleLoaderConfig}}}],lifecycleHooks:([] as any[])}}];
}
export function TestabilityNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.Testability,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgZone}}}],
      lifecycleHooks:([] as any[])}}];
}
export function TestabilityRegistryNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.TestabilityRegistry,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}];
}
export function ɵConsoleNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵConsole,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
