// @ng-bootstrap/ng-bootstrap/timepicker/timepicker.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker';
import * as i3 from '@angular/forms';
import * as i4 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker-config';
const styles_NgbTimepicker:any[] = ['.ngb-tp[_ngcontent-%COMP%] {\n      display: flex;\n      align-items: center;\n    }\n\n    .ngb-tp-hour[_ngcontent-%COMP%], .ngb-tp-minute[_ngcontent-%COMP%], .ngb-tp-second[_ngcontent-%COMP%], .ngb-tp-meridian[_ngcontent-%COMP%] {\n      display: flex;\n      flex-direction: column;\n      align-items: center;\n      justify-content: space-around;\n    }\n\n    .ngb-tp-spacer[_ngcontent-%COMP%] {\n      width: 1em;\n      text-align: center;\n    }\n\n    .chevron[_ngcontent-%COMP%]::before {\n      border-style: solid;\n      border-width: 0.29em 0.29em 0 0;\n      content: \'\';\n      display: inline-block;\n      height: 0.69em;\n      left: 0.05em;\n      position: relative;\n      top: 0.15em;\n      transform: rotate(-45deg);\n      -webkit-transform: rotate(-45deg);\n      -ms-transform: rotate(-45deg);\n      vertical-align: middle;\n      width: 0.71em;\n    }\n\n    .chevron.bottom[_ngcontent-%COMP%]:before {\n      top: -.3em;\n      -webkit-transform: rotate(135deg);\n      -ms-transform: rotate(135deg);\n      transform: rotate(135deg);\n    }\n\n    .btn-link[_ngcontent-%COMP%] {\n      outline: 0;\n    }\n\n    .btn-link.disabled[_ngcontent-%COMP%] {\n      cursor: not-allowed;\n      opacity: .65;\n    }\n\n    input[_ngcontent-%COMP%] {\n      text-align: center;\n      display: inline-block;\n      width: auto;\n    }'];
export const RenderType_NgbTimepicker:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_NgbTimepicker,data:{}});
function View_NgbTimepicker_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeHour(_co.hourStep)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Increment hours'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeHour((0 - _co.hourStep))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron bottom']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Decrement hours'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeMinute(_co.minuteStep)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Increment minutes'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeMinute((0 - _co.minuteStep))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron bottom']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Decrement minutes'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'ngb-tp-spacer']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),[':']))],(null as any),(null as any));
}
function View_NgbTimepicker_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeSecond(_co.secondStep)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Increment seconds'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'button',[['class',
      'btn-link'],['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.changeSecond((0 - _co.secondStep))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','chevron bottom']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Decrement seconds'])),(_l()(),i0.ɵted((null as any),['\n          ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = 'btn-link';
        const currVal_3:any = _co.setButtonSize();
        _ck(_v,1,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_NgbTimepicker_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'div',[['class',
      'ngb-tp-second']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_NgbTimepicker_7)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'input',[['aria-label','Seconds'],['class','form-control'],
              ['maxlength','2'],['placeholder','SS'],['size','2'],['type','text']],
          [[8,'value',0],[8,'readOnly',0],[8,'disabled',0]],[[(null as any),'change']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('change' === en)) {
              const pd_0:any = ((<any>_co.updateSecond($event.target.value)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_NgbTimepicker_8)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.spinners;
    _ck(_v,3,0,currVal_0);
    const currVal_4:any = 'form-control';
    const currVal_5:any = _co.setFormControlSize();
    _ck(_v,6,0,currVal_4,currVal_5);
    const currVal_6:any = _co.spinners;
    _ck(_v,9,0,currVal_6);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.formatMinSec(((_co.model == null)? (null as any): _co.model.second));
    const currVal_2:any = _co.readonlyInputs;
    const currVal_3:any = _co.disabled;
    _ck(_v,5,0,currVal_1,currVal_2,currVal_3);
  });
}
function View_NgbTimepicker_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'ngb-tp-spacer']],(null as any),(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(null as any));
}
function View_NgbTimepicker_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'div',[['class',
      'ngb-tp-meridian']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),2,'button',[['class','btn btn-outline-primary'],
          ['type','button']],[[8,'disabled',0],[2,'disabled',(null as any)]],[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.toggleMeridian()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),
      ['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = 'btn btn-outline-primary';
    const currVal_3:any = _co.setButtonSize();
    _ck(_v,3,0,currVal_2,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.disabled;
    const currVal_1:any = _co.disabled;
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_4:any = ((((_co.model == null)? (null as any): _co.model.hour) >= 12)? 'PM': 'AM');
    _ck(_v,4,0,currVal_4);
  });
}
export function View_NgbTimepicker_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),43,'fieldset',([] as any[]),[[8,'disabled',0],[2,'disabled',(null as any)]],
      (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
      ['\n      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),40,'div',[['class',
      'ngb-tp']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),10,'div',[['class','ngb-tp-hour']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_NgbTimepicker_1)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'input',[['aria-label','Hours'],['class','form-control'],
              ['maxlength','2'],['placeholder','HH'],['size','2'],['type','text']],
          [[8,'value',0],[8,'readOnly',0],[8,'disabled',0]],[[(null as any),'change']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i2.NgbTimepicker = _v.component;
            if (('change' === en)) {
              const pd_0:any = ((<any>_co.updateHour($event.target.value)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_NgbTimepicker_2)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class','ngb-tp-spacer']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),[':'])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),10,'div',[['class','ngb-tp-minute']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_NgbTimepicker_3)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'input',[['aria-label','Minutes'],['class','form-control'],
              ['maxlength','2'],['placeholder','MM'],['size','2'],['type','text']],
          [[8,'value',0],[8,'readOnly',0],[8,'disabled',0]],[[(null as any),'change']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i2.NgbTimepicker = _v.component;
            if (('change' === en)) {
              const pd_0:any = ((<any>_co.updateMinute($event.target.value)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_NgbTimepicker_4)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbTimepicker_5)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbTimepicker_6)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbTimepicker_9)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbTimepicker_10)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),
          i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n  ']))],
      (_ck,_v) => {
        var _co:i2.NgbTimepicker = _v.component;
        const currVal_2:any = _co.spinners;
        _ck(_v,8,0,currVal_2);
        const currVal_6:any = 'form-control';
        const currVal_7:any = _co.setFormControlSize();
        _ck(_v,11,0,currVal_6,currVal_7);
        const currVal_8:any = _co.spinners;
        _ck(_v,14,0,currVal_8);
        const currVal_9:any = _co.spinners;
        _ck(_v,23,0,currVal_9);
        const currVal_13:any = 'form-control';
        const currVal_14:any = _co.setFormControlSize();
        _ck(_v,26,0,currVal_13,currVal_14);
        const currVal_15:any = _co.spinners;
        _ck(_v,29,0,currVal_15);
        const currVal_16:any = _co.seconds;
        _ck(_v,33,0,currVal_16);
        const currVal_17:any = _co.seconds;
        _ck(_v,36,0,currVal_17);
        const currVal_18:any = _co.meridian;
        _ck(_v,39,0,currVal_18);
        const currVal_19:any = _co.meridian;
        _ck(_v,42,0,currVal_19);
      },(_ck,_v) => {
        var _co:i2.NgbTimepicker = _v.component;
        const currVal_0:any = _co.disabled;
        const currVal_1:any = _co.disabled;
        _ck(_v,1,0,currVal_0,currVal_1);
        const currVal_3:any = _co.formatHour(((_co.model == null)? (null as any): _co.model.hour));
        const currVal_4:any = _co.readonlyInputs;
        const currVal_5:any = _co.disabled;
        _ck(_v,10,0,currVal_3,currVal_4,currVal_5);
        const currVal_10:any = _co.formatMinSec(((_co.model == null)? (null as any): _co.model.minute));
        const currVal_11:any = _co.readonlyInputs;
        const currVal_12:any = _co.disabled;
        _ck(_v,25,0,currVal_10,currVal_11,currVal_12);
      });
}
export function View_NgbTimepicker_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ngb-timepicker',
      ([] as any[]),(null as any),(null as any),(null as any),View_NgbTimepicker_0,
      RenderType_NgbTimepicker)),i0.ɵprd(5120,(null as any),i3.NG_VALUE_ACCESSOR,(p0_0:any) => {
    return [p0_0];
  },[i2.NgbTimepicker]),i0.ɵdid(573440,(null as any),0,i2.NgbTimepicker,[i4.NgbTimepickerConfig],
      (null as any),(null as any))],(null as any),(null as any));
}
export const NgbTimepickerNgFactory:i0.ComponentFactory<i2.NgbTimepicker> = i0.ɵccf('ngb-timepicker',
    i2.NgbTimepicker,View_NgbTimepicker_Host_0,{meridian:'meridian',spinners:'spinners',
        seconds:'seconds',hourStep:'hourStep',minuteStep:'minuteStep',secondStep:'secondStep',
        readonlyInputs:'readonlyInputs',size:'size'},{},([] as any[]));
