// @ng-bootstrap/ng-bootstrap/rating/rating.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@ng-bootstrap/ng-bootstrap/rating/rating';
import * as i3 from '@angular/forms';
import * as i4 from '@ng-bootstrap/ng-bootstrap/rating/rating-config';
const styles_NgbRating:any[] = ([] as any[]);
export const RenderType_NgbRating:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_NgbRating,
    data:{}});
function View_NgbRating_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = ((_v.context.fill === 100)? '★': '☆');
    _ck(_v,0,0,currVal_0);
  });
}
function View_NgbRating_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbRating_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),1,'span',[['class','sr-only']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['(',
      ')'])),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),4,'span',([] as any[]),[[4,'cursor',(null as any)]],[[(null as any),
          'mouseenter'],[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('mouseenter' === en)) {
          const pd_0:any = ((<any>_co.enter((_v.context.index + 1))) !== false);
          ad = (pd_0 && ad);
        }
        if (('click' === en)) {
          const pd_1:any = ((<any>_co.handleClick((_v.context.index + 1))) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbRating_3)),
      i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutletContext:[0,
          'ngTemplateOutletContext'],ngTemplateOutlet:[1,'ngTemplateOutlet']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵted((null as any),['\n    ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = _co.contexts[_v.context.index];
        const currVal_3:any = (_co.starTemplate || i0.ɵnov((<any>_v.parent),1));
        _ck(_v,7,0,currVal_2,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = ((_v.context.index < _co.nextRate)? '*': ' ');
        _ck(_v,2,0,currVal_0);
        const currVal_1:any = ((_co.readonly || _co.disabled)? 'default': 'pointer');
        _ck(_v,4,0,currVal_1);
      });
}
export function View_NgbRating_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(0,[['t',
      2]],(null as any),0,(null as any),View_NgbRating_1)),(_l()(),i0.ɵted((null as any),
      ['\n    '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_NgbRating_2)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,
      i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),
      i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i2.NgbRating = _v.component;
    const currVal_0:any = _co.contexts;
    _ck(_v,4,0,currVal_0);
  },(null as any));
}
export function View_NgbRating_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'ngb-rating',[['aria-valuemin',
      '0'],['class','d-inline-flex'],['role','slider'],['tabindex','0']],[[1,'aria-valuemax',
      0],[1,'aria-valuenow',0],[1,'aria-valuetext',0],[1,'aria-disabled',0]],[[(null as any),
      'blur'],[(null as any),'keydown'],[(null as any),'mouseleave']],(_v,en,$event) => {
    var ad:boolean = true;
    if (('blur' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,2).handleBlur()) !== false);
      ad = (pd_0 && ad);
    }
    if (('keydown' === en)) {
      const pd_1:any = ((<any>i0.ɵnov(_v,2).handleKeyDown($event)) !== false);
      ad = (pd_1 && ad);
    }
    if (('mouseleave' === en)) {
      const pd_2:any = ((<any>i0.ɵnov(_v,2).reset()) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },View_NgbRating_0,RenderType_NgbRating)),i0.ɵprd(5120,(null as any),i3.NG_VALUE_ACCESSOR,
      (p0_0:any) => {
        return [p0_0];
      },[i2.NgbRating]),i0.ɵdid(638976,(null as any),1,i2.NgbRating,[i4.NgbRatingConfig,
      i0.ChangeDetectorRef],(null as any),(null as any)),i0.ɵqud(335544320,1,{starTemplate:0})],
      (_ck,_v) => {
        _ck(_v,2,0);
      },(_ck,_v) => {
        const currVal_0:any = i0.ɵnov(_v,2).max;
        const currVal_1:any = i0.ɵnov(_v,2).nextRate;
        const currVal_2:any = i0.ɵnov(_v,2).ariaValueText();
        const currVal_3:any = (i0.ɵnov(_v,2).readonly? true: (null as any));
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
      });
}
export const NgbRatingNgFactory:i0.ComponentFactory<i2.NgbRating> = i0.ɵccf('ngb-rating',
    i2.NgbRating,View_NgbRating_Host_0,{max:'max',rate:'rate',readonly:'readonly',
        resettable:'resettable',starTemplate:'starTemplate'},{hover:'hover',leave:'leave',
        rateChange:'rateChange'},([] as any[]));
