// @ng-bootstrap/ng-bootstrap/typeahead/typeahead-window.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from './highlight.ngfactory';
import * as i2 from '@ng-bootstrap/ng-bootstrap/typeahead/highlight';
import * as i3 from '@angular/common';
import * as i4 from '@ng-bootstrap/ng-bootstrap/typeahead/typeahead-window';
const styles_NgbTypeaheadWindow:any[] = ([] as any[]);
export const RenderType_NgbTypeaheadWindow:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_NgbTypeaheadWindow,data:{}});
function View_NgbTypeaheadWindow_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),1,'ngb-highlight',([] as any[]),(null as any),(null as any),
      (null as any),i1.View_NgbHighlight_0,i1.RenderType_NgbHighlight)),i0.ɵdid(573440,
      (null as any),0,i2.NgbHighlight,([] as any[]),{result:[0,'result'],term:[1,'term']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    const currVal_0:any = _v.context.formatter(_v.context.result);
    const currVal_1:any = _v.context.term;
    _ck(_v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_NgbTypeaheadWindow_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbTypeaheadWindow_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),5,'button',[['class','dropdown-item'],['role','option'],
          ['type','button']],[[8,'id',0],[2,'active',(null as any)]],[[(null as any),
          'mouseenter'],[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('mouseenter' === en)) {
          const pd_0:any = ((<any>_co.markActive(_v.context.index)) !== false);
          ad = (pd_0 && ad);
        }
        if (('click' === en)) {
          const pd_1:any = ((<any>_co.select(_v.context.$implicit)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),2,(null as any),View_NgbTypeaheadWindow_3)),
      i0.ɵdid(540672,(null as any),0,i3.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutletContext:[0,
          'ngTemplateOutletContext'],ngTemplateOutlet:[1,'ngTemplateOutlet']},(null as any)),
      i0.ɵpod({result:0,term:1,formatter:2}),(_l()(),i0.ɵted((null as any),['\n      '])),
      (_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = _ck(_v,5,0,_v.context.$implicit,_co.term,_co.formatter);
    const currVal_3:any = (_co.resultTemplate || i0.ɵnov((<any>_v.parent),1));
    _ck(_v,4,0,currVal_2,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((_co.id + '-') + _v.context.index);
    const currVal_1:any = (_v.context.index === _co.activeIdx);
    _ck(_v,1,0,currVal_0,currVal_1);
  });
}
export function View_NgbTypeaheadWindow_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(0,[['rt',
      2]],(null as any),0,(null as any),View_NgbTypeaheadWindow_1)),(_l()(),i0.ɵted((null as any),
      ['\n    '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_NgbTypeaheadWindow_2)),i0.ɵdid(802816,(null as any),0,i3.NgForOf,[i0.ViewContainerRef,
      i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),
      i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i4.NgbTypeaheadWindow = _v.component;
    const currVal_0:any = _co.results;
    _ck(_v,4,0,currVal_0);
  },(null as any));
}
export function View_NgbTypeaheadWindow_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ngb-typeahead-window',
      [['class','dropdown-menu'],['role','listbox'],['style','display: block']],[[8,
          'id',0]],(null as any),(null as any),View_NgbTypeaheadWindow_0,RenderType_NgbTypeaheadWindow)),
      i0.ɵdid(114688,(null as any),0,i4.NgbTypeaheadWindow,([] as any[]),(null as any),
          (null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).id;
    _ck(_v,0,0,currVal_0);
  });
}
export const NgbTypeaheadWindowNgFactory:i0.ComponentFactory<i4.NgbTypeaheadWindow> = i0.ɵccf('ngb-typeahead-window',
    i4.NgbTypeaheadWindow,View_NgbTypeaheadWindow_Host_0,{id:'id',focusFirst:'focusFirst',
        results:'results',term:'term',formatter:'formatter',resultTemplate:'resultTemplate'},
    {selectEvent:'select',activeChangeEvent:'activeChange'},([] as any[]));
