// ng-jhipster/src/component/jhi-item-count.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '../language/jhi-translate.directive.ngfactory';
import * as i2 from 'ng-jhipster/src/language/jhi-translate.directive';
import * as i3 from '@angular/common';
import * as i4 from 'ng-jhipster/src/component/jhi-item-count.component';
import * as i5 from 'ng-jhipster/src/config.service';
const styles_JhiItemCountComponent:any[] = ([] as any[]);
export const RenderType_JhiItemCountComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_JhiItemCountComponent,data:{}});
function View_JhiItemCountComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'info jhi-item-count'],['jhiTranslate','global.item-count']],[[1,'translateValues',
      0]],(null as any),(null as any),i1.View_JhiTranslateComponent_0,i1.RenderType_JhiTranslateComponent)),
      i0.ɵdid(49152,(null as any),0,i2.JhiTranslateComponent,([] as any[]),{jhiTranslate:[0,
          'jhiTranslate'],translateValues:[1,'translateValues']},(null as any)),(_l()(),
          i0.ɵted((null as any),['  /* [attr.translateValues] is used to get entire values in tests */\n        ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = 'global.item-count';
        const currVal_2:any = i0.ɵinlineInterpolate(1,'',_co.i18nValues(),'');
        _ck(_v,1,0,currVal_1,currVal_2);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.i18nValues();
        _ck(_v,0,0,currVal_0);
      });
}
function View_JhiItemCountComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n            Showing ',' -\n            ',
      '\n            of ',' items.\n        ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((((_co.page - 1) * _co.itemsPerPage) == 0)? 1: (((_co.page - 1) * _co.itemsPerPage) + 1));
    const currVal_1:any = (((_co.page * _co.itemsPerPage) < _co.total)? (_co.page * _co.itemsPerPage): _co.total);
    const currVal_2:any = _co.total;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_JhiItemCountComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_JhiItemCountComponent_1)),i0.ɵdid(16384,
      (null as any),0,i3.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf'],
          ngIfElse:[1,'ngIfElse']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(0,[['noI18n',2]],(null as any),0,(null as any),View_JhiItemCountComponent_2))],
      (_ck,_v) => {
        var _co:i4.JhiItemCountComponent = _v.component;
        const currVal_0:any = _co.i18nEnabled;
        const currVal_1:any = i0.ɵnov(_v,4);
        _ck(_v,2,0,currVal_0,currVal_1);
      },(null as any));
}
export function View_JhiItemCountComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'jhi-item-count',
      ([] as any[]),(null as any),(null as any),(null as any),View_JhiItemCountComponent_0,
      RenderType_JhiItemCountComponent)),i0.ɵdid(49152,(null as any),0,i4.JhiItemCountComponent,
      [i5.JhiConfigService],(null as any),(null as any))],(null as any),(null as any));
}
export const JhiItemCountComponentNgFactory:i0.ComponentFactory<i4.JhiItemCountComponent> = i0.ɵccf('jhi-item-count',
    i4.JhiItemCountComponent,View_JhiItemCountComponent_Host_0,{page:'page',total:'total',
        itemsPerPage:'itemsPerPage'},{},([] as any[]));
