// @ng-bootstrap/ng-bootstrap/tabset/tabset.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@ng-bootstrap/ng-bootstrap/tabset/tabset';
import * as i3 from '@ng-bootstrap/ng-bootstrap/tabset/tabset-config';
const styles_NgbTabset:any[] = ([] as any[]);
export const RenderType_NgbTabset:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_NgbTabset,
    data:{}});
function View_NgbTabset_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbTabset_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'nav-item']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),4,'a',[['class','nav-link'],['href',''],['role','tab']],[[8,
              'id',0],[2,'active',(null as any)],[2,'disabled',(null as any)],[1,'tabindex',
              0],[1,'aria-controls',0],[1,'aria-expanded',0],[1,'aria-disabled',0]],
          [[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>!!_co.select(_v.context.$implicit.id)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          ',
          ''])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_NgbTabset_2)),i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,[i0.ViewContainerRef],
          {ngTemplateOutlet:[0,'ngTemplateOutlet']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n      ']))],(_ck,_v) => {
    const currVal_8:any = ((_v.context.$implicit.titleTpl == null)? (null as any): _v.context.$implicit.titleTpl.templateRef);
    _ck(_v,5,0,currVal_8);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _v.context.$implicit.id;
    const currVal_1:any = (_v.context.$implicit.id === _co.activeId);
    const currVal_2:any = _v.context.$implicit.disabled;
    const currVal_3:any = (_v.context.$implicit.disabled? '-1': (undefined as any));
    const currVal_4:any = ((!_co.destroyOnHide || (_v.context.$implicit.id === _co.activeId))? (_v.context.$implicit.id + '-panel'): (null as any));
    const currVal_5:any = (_v.context.$implicit.id === _co.activeId);
    const currVal_6:any = _v.context.$implicit.disabled;
    _ck(_v,2,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_7:any = _v.context.$implicit.title;
    _ck(_v,3,0,currVal_7);
  });
}
function View_NgbTabset_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbTabset_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',[['role',
      'tabpanel']],[[8,'className',0],[1,'aria-labelledby',0],[8,'id',0],[1,'aria-expanded',
      0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
      ['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_NgbTabset_5)),i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,[i0.ViewContainerRef],
      {ngTemplateOutlet:[0,'ngTemplateOutlet']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n        ']))],(_ck,_v) => {
    const currVal_4:any = (<any>_v.parent).context.$implicit.contentTpl.templateRef;
    _ck(_v,3,0,currVal_4);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'tab-pane ',(((<any>_v.parent).context.$implicit.id === _co.activeId)? 'active': (null as any)),
        '');
    const currVal_1:any = (<any>_v.parent).context.$implicit.id;
    const currVal_2:any = i0.ɵinlineInterpolate(1,'',(<any>_v.parent).context.$implicit.id,
        '-panel');
    const currVal_3:any = ((<any>_v.parent).context.$implicit.id === _co.activeId);
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_NgbTabset_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_NgbTabset_4)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = (!_co.destroyOnHide || (_v.context.$implicit.id === _co.activeId));
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_NgbTabset_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),4,'ul',[['role','tablist']],[[8,'className',0]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n      '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbTabset_1)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n    '])),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),4,'div',[['class','tab-content']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_NgbTabset_3)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,
          i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n  ']))],
      (_ck,_v) => {
        var _co:i2.NgbTabset = _v.component;
        const currVal_1:any = _co.tabs;
        _ck(_v,4,0,currVal_1);
        const currVal_2:any = _co.tabs;
        _ck(_v,10,0,currVal_2);
      },(_ck,_v) => {
        var _co:i2.NgbTabset = _v.component;
        const currVal_0:any = (('nav nav-' + _co.type) + ((_co.orientation == 'horizontal')? (' ' + _co.justifyClass): ' flex-column'));
        _ck(_v,1,0,currVal_0);
      });
}
export function View_NgbTabset_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ngb-tabset',([] as any[]),
      (null as any),(null as any),(null as any),View_NgbTabset_0,RenderType_NgbTabset)),
      i0.ɵdid(2146304,(null as any),1,i2.NgbTabset,[i3.NgbTabsetConfig],(null as any),
          (null as any)),i0.ɵqud(603979776,1,{tabs:1})],(null as any),(null as any));
}
export const NgbTabsetNgFactory:i0.ComponentFactory<i2.NgbTabset> = i0.ɵccf('ngb-tabset',
    i2.NgbTabset,View_NgbTabset_Host_0,{activeId:'activeId',destroyOnHide:'destroyOnHide',
        justify:'justify',orientation:'orientation',type:'type'},{tabChange:'tabChange'},
    ([] as any[]));
