// @ng-bootstrap/ng-bootstrap/pagination/pagination.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@ng-bootstrap/ng-bootstrap/pagination/pagination';
import * as i3 from '@ng-bootstrap/ng-bootstrap/pagination/pagination-config';
const styles_NgbPagination:any[] = ([] as any[]);
export const RenderType_NgbPagination:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_NgbPagination,data:{}});
function View_NgbPagination_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'page-item']],[[2,'disabled',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),4,'a',[['aria-label','First'],['class','page-link'],
          ['href','']],[[1,'tabindex',0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>!!_co.selectPage(1)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['aria-hidden','true']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['««'])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = (!_co.hasPrevious() || _co.disabled);
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = (_co.hasPrevious()? (null as any): '-1');
    _ck(_v,2,0,currVal_1);
  });
}
function View_NgbPagination_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'page-item']],[[2,'disabled',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),4,'a',[['aria-label','Previous'],['class','page-link'],
          ['href','']],[[1,'tabindex',0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>!!_co.selectPage((_co.page - 1))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['aria-hidden','true']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['«'])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = (!_co.hasPrevious() || _co.disabled);
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = (_co.hasPrevious()? (null as any): '-1');
    _ck(_v,2,0,currVal_1);
  });
}
function View_NgbPagination_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'a',[['class',
      'page-link']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['...']))],(null as any),(null as any));
}
function View_NgbPagination_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'sr-only']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['(current)']))],(null as any),(null as any));
}
function View_NgbPagination_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['class',
      'page-link'],['href','']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>!!_co.selectPage((<any>_v.parent).context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          ','\n          '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_6)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = ((<any>_v.parent).context.$implicit === _co.page);
        _ck(_v,3,0,currVal_1);
      },(_ck,_v) => {
        const currVal_0:any = (<any>_v.parent).context.$implicit;
        _ck(_v,1,0,currVal_0);
      });
}
function View_NgbPagination_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'page-item']],[[2,'active',(null as any)],[2,'disabled',(null as any)]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_4)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_5)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      ']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_2:any = _co.isEllipsis(_v.context.$implicit);
    _ck(_v,3,0,currVal_2);
    const currVal_3:boolean = !_co.isEllipsis(_v.context.$implicit);
    _ck(_v,6,0,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_v.context.$implicit === _co.page);
    const currVal_1:any = (_co.isEllipsis(_v.context.$implicit) || _co.disabled);
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbPagination_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'page-item']],[[2,'disabled',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),4,'a',[['aria-label','Next'],['class','page-link'],
          ['href','']],[[1,'tabindex',0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>!!_co.selectPage((_co.page + 1))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['aria-hidden','true']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['»'])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = (!_co.hasNext() || _co.disabled);
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = (_co.hasNext()? (null as any): '-1');
    _ck(_v,2,0,currVal_1);
  });
}
function View_NgbPagination_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'page-item']],[[2,'disabled',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),4,'a',[['aria-label','Last'],['class','page-link'],
          ['href','']],[[1,'tabindex',0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>!!_co.selectPage(_co.pageCount)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['aria-hidden','true']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['»»'])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = (!_co.hasNext() || _co.disabled);
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = (_co.hasNext()? (null as any): '-1');
    _ck(_v,2,0,currVal_1);
  });
}
export function View_NgbPagination_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),16,'ul',([] as any[]),[[8,'className',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_1)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n\n      '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_2)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbPagination_3)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_NgbPagination_7)),i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n\n      '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
          (null as any),View_NgbPagination_8)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n  ']))],
      (_ck,_v) => {
        var _co:i2.NgbPagination = _v.component;
        const currVal_1:any = _co.boundaryLinks;
        _ck(_v,4,0,currVal_1);
        const currVal_2:any = _co.directionLinks;
        _ck(_v,7,0,currVal_2);
        const currVal_3:any = _co.pages;
        _ck(_v,10,0,currVal_3);
        const currVal_4:any = _co.directionLinks;
        _ck(_v,13,0,currVal_4);
        const currVal_5:any = _co.boundaryLinks;
        _ck(_v,16,0,currVal_5);
      },(_ck,_v) => {
        var _co:i2.NgbPagination = _v.component;
        const currVal_0:any = ('pagination' + (_co.size? (' pagination-' + _co.size): ''));
        _ck(_v,1,0,currVal_0);
      });
}
export function View_NgbPagination_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ngb-pagination',
      [['role','navigation']],(null as any),(null as any),(null as any),View_NgbPagination_0,
      RenderType_NgbPagination)),i0.ɵdid(573440,(null as any),0,i2.NgbPagination,[i3.NgbPaginationConfig],
      (null as any),(null as any))],(null as any),(null as any));
}
export const NgbPaginationNgFactory:i0.ComponentFactory<i2.NgbPagination> = i0.ɵccf('ngb-pagination',
    i2.NgbPagination,View_NgbPagination_Host_0,{disabled:'disabled',boundaryLinks:'boundaryLinks',
        directionLinks:'directionLinks',ellipses:'ellipses',rotate:'rotate',collectionSize:'collectionSize',
        maxSize:'maxSize',page:'page',pageSize:'pageSize',size:'size'},{pageChange:'pageChange'},
    ([] as any[]));
