// @angular/common/common.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/common';
import * as i1 from '@angular/core';
export function CommonModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.NgLocalization}},
          useClass:{reference:i0.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
          multi:false},module:{reference:i0.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i0.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.NgClass},{reference:i0.NgComponentOutlet},
          {reference:i0.NgForOf},{reference:i0.NgIf},{reference:i0.NgTemplateOutlet},
          {reference:i0.NgStyle},{reference:i0.NgSwitch},{reference:i0.NgSwitchCase},
          {reference:i0.NgSwitchDefault},{reference:i0.NgPlural},{reference:i0.NgPluralCase}],
      exportedPipes:[{reference:i0.AsyncPipe},{reference:i0.UpperCasePipe},{reference:i0.LowerCasePipe},
          {reference:i0.JsonPipe},{reference:i0.SlicePipe},{reference:i0.DecimalPipe},
          {reference:i0.PercentPipe},{reference:i0.TitleCasePipe},{reference:i0.CurrencyPipe},
          {reference:i0.DatePipe},{reference:i0.I18nPluralPipe},{reference:i0.I18nSelectPipe}]},
      NgClassNgSummary,NgComponentOutletNgSummary,NgForOfNgSummary,NgIfNgSummary,NgTemplateOutletNgSummary,
      NgStyleNgSummary,NgSwitchNgSummary,NgSwitchCaseNgSummary,NgSwitchDefaultNgSummary,
      NgPluralNgSummary,NgPluralCaseNgSummary,AsyncPipeNgSummary,UpperCasePipeNgSummary,
      LowerCasePipeNgSummary,JsonPipeNgSummary,SlicePipeNgSummary,DecimalPipeNgSummary,
      PercentPipeNgSummary,TitleCasePipeNgSummary,CurrencyPipeNgSummary,DatePipeNgSummary,
      I18nPluralPipeNgSummary,I18nSelectPipeNgSummary,{summaryKind:3,type:{reference:i0.NgLocaleLocalization,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.LOCALE_ID}}}],lifecycleHooks:([] as any[])}}];
}
export function NgClassNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgClass,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.IterableDiffers}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.KeyValueDiffers}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.Renderer}}}],lifecycleHooks:[2]},isComponent:false,
      selector:'[ngClass]',exportAs:(null as any),inputs:{klass:'class',ngClass:'ngClass'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgComponentOutletNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgComponentOutlet,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}}],
      lifecycleHooks:[1,3]},isComponent:false,selector:'[ngComponentOutlet]',exportAs:(null as any),
      inputs:{ngComponentOutlet:'ngComponentOutlet',ngComponentOutletInjector:'ngComponentOutletInjector',
          ngComponentOutletContent:'ngComponentOutletContent',ngComponentOutletNgModuleFactory:'ngComponentOutletNgModuleFactory'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgForOfNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgForOf,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.TemplateRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.IterableDiffers}}}],
      lifecycleHooks:[2,3]},isComponent:false,selector:'[ngFor][ngForOf]',exportAs:(null as any),
      inputs:{ngForOf:'ngForOf',ngForTrackBy:'ngForTrackBy',ngForTemplate:'ngForTemplate'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgIfNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgIf,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.TemplateRef}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[ngIf]',exportAs:(null as any),inputs:{ngIf:'ngIf',
          ngIfThen:'ngIfThen',ngIfElse:'ngIfElse'},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function NgTemplateOutletNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgTemplateOutlet,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}}],
      lifecycleHooks:[3]},isComponent:false,selector:'[ngTemplateOutlet]',exportAs:(null as any),
      inputs:{ngTemplateOutletContext:'ngTemplateOutletContext',ngTemplateOutlet:'ngTemplateOutlet',
          ngOutletContext:'ngOutletContext'},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function NgStyleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgStyle,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.KeyValueDiffers}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.Renderer}}}],
      lifecycleHooks:[2]},isComponent:false,selector:'[ngStyle]',exportAs:(null as any),
      inputs:{ngStyle:'ngStyle'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgSwitchNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgSwitch,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[ngSwitch]',exportAs:(null as any),inputs:{ngSwitch:'ngSwitch'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgSwitchCaseNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgSwitchCase,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.TemplateRef}}},{isAttribute:false,isHost:true,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgSwitch}}}],
      lifecycleHooks:[2]},isComponent:false,selector:'[ngSwitchCase]',exportAs:(null as any),
      inputs:{ngSwitchCase:'ngSwitchCase'},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function NgSwitchDefaultNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgSwitchDefault,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.TemplateRef}}},{isAttribute:false,isHost:true,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgSwitch}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[ngSwitchDefault]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function NgPluralNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgPlural,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgLocalization}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[ngPlural]',exportAs:(null as any),
      inputs:{ngPlural:'ngPlural'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function NgPluralCaseNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.NgPluralCase,diDeps:[{isAttribute:true,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{value:'ngPluralCase'}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.TemplateRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ViewContainerRef}}},
      {isAttribute:false,isHost:true,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.NgPlural}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[ngPluralCase]',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function AsyncPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.AsyncPipe,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ChangeDetectorRef}}}],
      lifecycleHooks:[1]},name:'async',pure:false}];
}
export function UpperCasePipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.UpperCasePipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'uppercase',pure:true}];
}
export function LowerCasePipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.LowerCasePipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'lowercase',pure:true}];
}
export function JsonPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.JsonPipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'json',pure:false}];
}
export function SlicePipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.SlicePipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'slice',pure:false}];
}
export function DecimalPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.DecimalPipe,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
      lifecycleHooks:([] as any[])},name:'number',pure:true}];
}
export function PercentPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.PercentPipe,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
      lifecycleHooks:([] as any[])},name:'percent',pure:true}];
}
export function TitleCasePipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.TitleCasePipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'titlecase',pure:true}];
}
export function CurrencyPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.CurrencyPipe,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
      lifecycleHooks:([] as any[])},name:'currency',pure:true}];
}
export function DatePipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.DatePipe,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
      lifecycleHooks:([] as any[])},name:'date',pure:true}];
}
export function I18nPluralPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.I18nPluralPipe,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.NgLocalization}}}],
      lifecycleHooks:([] as any[])},name:'i18nPlural',pure:true}];
}
export function I18nSelectPipeNgSummary():any[] {
  return [{summaryKind:0,type:{reference:i0.I18nSelectPipe,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      name:'i18nSelect',pure:true}];
}
export function HashLocationStrategyNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.HashLocationStrategy,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.PlatformLocation}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.APP_BASE_HREF}}}],lifecycleHooks:([] as any[])}}];
}
export function PathLocationStrategyNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.PathLocationStrategy,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.PlatformLocation}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.APP_BASE_HREF}}}],lifecycleHooks:([] as any[])}}];
}
export function LocationNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.Location,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.LocationStrategy}}}],
      lifecycleHooks:([] as any[])}}];
}
export function NgLocaleLocalizationNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.NgLocaleLocalization,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.LOCALE_ID}}}],
      lifecycleHooks:([] as any[])}}];
}
